tests/test_gs.py

```diff
@@ -0,0 +1,61 @@
+import unittest
+import os
+
+from to_data_library.data.gs import Client
+from tests.setup import setup
+from google.cloud import storage
+
+
+def setUpModule():
+    setup.create_bucket()
+
+
+def tearDownModule():
+    setup.delete_bucket()
+    setup.cleanup()
+
+
+class TestGS(unittest.TestCase):
+
+    def __init__(self, *args, **kwargs):
+        super(TestGS, self).__init__(*args, **kwargs)
+        self.setup = setup
+        self.gs_uri = f'gs://{self.setup.bucket_name}/sample.csv'
+        
+    def test_download(self):
```

Comment:
I used the Python Mocking library last week and was doing some research so forgive me for this, it's just fresh in my mind :)  This test seems to cover the functionality of the underlying library which should already be tested by Google. 

Is there much value in uploading and downloading files?  Could we mock instead as the storage library should be expected to work.   We could mock the outside functionality and it will speed up tests and only test the function itself.

e.g.  ( I haven't tested this, I'm also used to pytest)

```
from unittest.mock import patch, Mock 



class TestClient(unittest.TestCase): 
  @patch('to_data_library.data.gs,storage')
  @patch('builtins.open', mock_open())
  def test_download(self, mock_file, MockStorage):
     MockStorage = Mock()
     MockStorage.download_blob_to_file.return_value = Mock()


     client = Client('test-project')
     client.download('gs://bucket/file', 'dest_file')
     
     mock_file.assert_called_once_with('dest_file', 'wb')
     MockStorage.download_blob_to_file.assert_called_once_with('gs://bucket/file', mock_file.return_value.__enter__.return_value)
```

Reply:
though maybe it makes it more complex to test.